jina/serve/gateway.py

```diff
@@ -1,18 +1,89 @@
 import abc
 import argparse
-from typing import TYPE_CHECKING, Optional
+import functools
+import inspect
+from typing import TYPE_CHECKING, Callable, Optional
 
+from jina.helper import convert_tuple_to_list
 from jina.jaml import JAMLCompatible
 from jina.logging.logger import JinaLogger
+from jina.serve.helper import wrap_func
 from jina.serve.streamer import GatewayStreamer
 
 __all__ = ['BaseGateway']
 
 if TYPE_CHECKING:
     from prometheus_client import CollectorRegistry
 
+# TODO make this common with jina/serve/executors/decorators.py
+def store_init_kwargs(func: Callable) -> Callable:
```

Comment:
what is the differecne with the `Executor` one?

Reply:
minor differences: gateway does not have metas, requests and runtime args like executor + different error message
But I agree common code should be re-used I already add it as todo